src/QGCApplication.h

```diff
@@ -57,9 +58,15 @@ class QGCApplication : public QApplication
 {
     Q_OBJECT
 public:
+
     QGCApplication(int &argc, char* argv[], bool unitTesting);
     ~QGCApplication();
 
+    DataClass* getDataClass(void){
+        if(!_dataClass) _dataClass = new DataClass();
```

Comment:
Where is this deleted? If you're using C++11 and beyond, would be better to use unique/shared ptr